glidertest/tools.py

```diff
@@ -847,41 +875,73 @@ def plot_ts_histograms(ds: xr.Dataset, ax: plt.Axes = None, **kw: dict) -> tuple
     SA = gsw.SA_from_SP(sal, depth, long, lat)
     CT = gsw.CT_from_t(SA, temp, depth)
 
-    # Reduce to middle 95% of values
-    temp_filtered = CT[(CT >= np.nanpercentile(CT, 2.5)) & (CT <= np.nanpercentile(CT, 97.5))]
-    sal_filtered = SA[(SA >= np.nanpercentile(SA, 2.5)) & (SA <= np.nanpercentile(SA, 97.5))]
+    # Reduce to middle 99% of values
+    CT_filtered = CT[(CT >= np.nanpercentile(CT, .5)) & (CT <= np.nanpercentile(CT, 99.5))]
+    SA_filtered = SA[(SA >= np.nanpercentile(SA, .5)) & (SA <= np.nanpercentile(SA, 99.5))]
+    print('Temperature and Salinity values have been filtered to the middle 99% of values.')
 
-    ax[0].hist(temp_filtered, bins=50, **kw)
+    ax[0].hist(CT_filtered, bins=num_bins, **kw)
     ax[0].set_xlabel('Conservative Temperature (°C)')
     ax[0].set_ylabel('Frequency')
     ax[0].set_title('Histogram of Temperature')
     ax[0].grid()
+    ax[0].set_xlim(CT_filtered.min(), CT_filtered.max())
 
-    ax[1].hist(sal_filtered, bins=50, **kw)
+    ax[1].hist(SA_filtered, bins=num_bins, **kw)
     ax[1].set_xlabel('Absolute Salinity ( )')
     ax[1].set_ylabel('Frequency')
     ax[1].set_title('Histogram of Salinity')
     ax[1].grid()
+    ax[1].set_xlim(SA_filtered.min(), SA_filtered.max())
 
-    h = ax[2].hist2d(sal, temp, bins=50, cmap='viridis', norm=mcolors.LogNorm(), **kw)
+    h = ax[2].hist2d(SA_filtered, CT_filtered, bins=num_bins, cmap='viridis', norm=mcolors.LogNorm(), **kw)
     cbar = plt.colorbar(h[3], ax=ax[2])
     cbar.set_label('Log Counts')
     ax[2].set_xlabel('Absolute Salinity ( )')
     ax[2].set_ylabel('Conservative Temperature (°C)')
-    ax[2].set_title('2D Histogram of Salinity and Temperature (Log Scale)')
+    ax[2].set_title('2D Histogram (Log Scale)')
+    # Set x-limits based on salinity plot and y-limits based on temperature plot
+    ax[2].set_xlim(ax[1].get_xlim())
+    ax[2].set_ylim(ax[0].get_xlim())
 
     # Calculate density and add contours
-    SA = gsw.SA_from_SP(sal, depth, long, lat)
-    CT = gsw.CT_from_t(SA, temp, depth)
-    sigma0 = gsw.sigma0(SA, CT)
-
-    xi = np.linspace(sal.min()-1, sal.max()+1, 100)
-    yi = np.linspace(temp.min()-1, temp.max()+1, 100)
+    xi = np.linspace(SA_filtered.min()-.2, SA_filtered.max()+.2, 100)

```

Comment:
The tests fail here because `np.linspace` doesn't like being passed xarray DataSets. Try just passing it the numpy array of values for `SA_filtered` and `CT_filtered` with:

```python
    xi = np.linspace(SA_filtered.values.min()-.2, SA_filtered.values.max()+.2, 100)
    yi = np.linspace(CT_filtered.values.min()-.2, CT_filtered.values.max()+.2, 100)
```

Reply:
OK!